plugins/modules/ndo_ipsla_monitoring_policy.py

```diff
@@ -0,0 +1,433 @@
+  req_data_size:
+    description:
+    - The size of the request data.
+    type: int
+    default: 28
+  type_of_service:
+    description:
+    - The IPv4 Type of Service.
+    type: int
+    default: 0
+  operation_timeout:
+    description:
+    - The operation_timeout for SLA.
+    type: int
+    default: 900
+  threshold:
+    description:
+    - The threshold for SLA.
+    type: int
+    default: 900
+  ipv6_traffic_class:
+    description:
+    - The IPv6 Traffic Class.
+    type: int
+    default: 0
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a new IPSLA monitoring policy
+  cisco.mso.ndo_ipsla_monitoring_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_monitoring_policy: ansible_test_ipsla_monitoring_policy
+    sla_type: http
+    http_version: HTTP11
+    http_uri: /test
+    sla_frequency: 120
+    detect_multiplier: 5
+    req_data_size: 64
+    type_of_service: 16
+    operation_timeout: 1000
+    threshold: 1000
+    ipv6_traffic_class: 32
+    state: present
+
+- name: Query an IPSLA monitoring policy with template_name
+  cisco.mso.ndo_ipsla_monitoring_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_monitoring_policy: ansible_test_ipsla_monitoring_policy
+    state: query
+  register: query_one
+
+- name: Query all IPSLA monitoring policies in the template
+  cisco.mso.ndo_ipsla_monitoring_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    state: query
+  register: query_all
+
+- name: Delete an IPSLA monitoring policy
+  cisco.mso.ndo_ipsla_monitoring_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    ipsla_monitoring_policy: ansible_test_ipsla_monitoring_policy
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import (
+    MSOModule,
+    mso_argument_spec,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.template import (
+    MSOTemplate,
+    KVPair,
+)
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True),
+        ipsla_monitoring_policy=dict(type="str", aliases=["name"]),
+        ipsla_monitoring_policy_uuid=dict(type="str", aliases=["uuid"]),
+        sla_type=dict(type="str", choices=["http", "tcp", "icmp", "l2ping"], default="icmp"),
+        destination_port=dict(type="int", default=0),
```

Comment:
do we need defaults for this? since typically we do not define defaults.

Reply:
Removed them